--- common/src/main/java/one/pkg/kreno/shared/network/TrafficMonitor.java
+++ common/src/main/java/one/pkg/kreno/shared/network/TrafficMonitor.java
@@ -62,6 +62,7 @@
             pPacketStat.bytes.addAndGet(bytes);
             pStat.totalIn.addAndGet(bytes);
         }
+        updateRates();
     }

     public static void onOutboundPacket(UUID playerUuid, String playerName, String packetName, int bytes) {
@@ -79,6 +80,7 @@
             pPacketStat.bytes.addAndGet(bytes);
             pStat.totalOut.addAndGet(bytes);
         }
+        updateRates();
     }

     public static void onInboundCompressed(int bytes) {
